src/17D6A0.c

```diff
@@ -94,11 +98,109 @@ void free_popup(PopupMessage* popup) {
 
 INCLUDE_ASM(s32, "17D6A0", func_8024EFE0);
 
-INCLUDE_ASM(s32, "17D6A0", func_8024F394);
+void func_8024F394(void* data) {
+    PopupMessage* popup = data;
+    Message* message = popup->message;
+    s32 found = FALSE;
+    s32 i;
+
+    for (i = 0; i < popup->messageIndex; i++, message++) {
```

Comment:
I think we should initialise `message` here too if we're going to be incrementing it here also
```suggestion
    for (i = 0, message = popup->message; i < popup->messageIndex; i++, message++) {
```

Reply:
I disagree - I think it makes the loop confusing to grok, and we have dozens of other loops like this in the game for which we don't initialize the struct pointer there either